src/lib.rs

```diff
@@ -163,6 +165,29 @@ impl Order {
         self.account.get(&mut self.nonce, challenge_url).await
     }
 
+    /// Poll the order with exponential backoff until in a final state
+    ///
+    /// Refresh the order state from the server for `tries` times, waiting `delay` before the
+    /// first attempt and increasing the delay by a factor of 2 for each subsequent attempt.
+    ///
+    /// Yields the [`OrderStatus`] immediately if final or after `tries` attempts.
+    ///
+    /// (Empirically, we've had good results with 5 tries and an initial delay of 250ms.)
+    pub async fn poll(&mut self, mut tries: u8, mut delay: Duration) -> Result<OrderStatus, Error> {
+        loop {
+            sleep(delay).await;
```

Comment:
I would have expected this to `self.refresh().await?` once up-front before sleeping. Was always sleeping for the `delay` before the first poll something you wanted to do on purpose?

Reply:
> (but I admit this is a bit corner-case-y).

Probably best to leave the initial sleep since that's most relevant to the vast majority of cases and accept that if you're doing weird stuff you might have to eat one extra sleep. 

I'll resolve the thread :+1: